great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -819,16 +821,36 @@ def qualified_name(self) -> str:
     @pydantic.validator("table_name", pre=True, always=True)
     def _default_table_name(cls, table_name: str, values: dict, **kwargs) -> str:
         if not (validated_table_name := table_name or values.get("name")):
+            # TODO: if using `name` we should replace spaces with underscores
+            # (unless bracketed by quotes)
             raise ValueError(
                 "table_name cannot be empty and should default to name if not provided"
             )
 
+        return validated_table_name
+
+    @pydantic.validator("table_name")
+    def _resolve_qouted_name(cls, table_name: str) -> str | quoted_name:
+        table_name_is_qouted: bool = cls._is_bracketed_by_quotes(table_name)
+
         from great_expectations.compatibility import sqlalchemy
 
         if sqlalchemy.quoted_name:
-            return sqlalchemy.quoted_name(value=validated_table_name, quote=True)
+            if isinstance(table_name, sqlalchemy.quoted_name):
+                return table_name
+
+            if table_name_is_qouted:
```

Comment:
nit - a few instances of "quoted" are misspelled in this PR (not a big issue but would be nice to clean up 😄 )

Reply:
Thanks, updated!